py-polars/tests/unit/test_queries.py

```diff
@@ -364,3 +364,30 @@ def test_datetime_supertype_5236() -> None:
     )
     assert out.shape == (0, 2)
     assert out.dtypes == [pl.Datetime("ns", "UTC")] * 2
+
+
+def test_utf8_date() -> None:
+    df = pl.DataFrame({"x1": ["2021-01-01"]}).with_columns(
+        **{"x1-date": pl.col("x1").cast(pl.Date)}
+    )
+    out = df.select(pl.col("x1-date"))
+    assert out.shape == (1, 1)
+    assert out.dtypes == [pl.Date]
```

Comment:
Please use the `assert_frame_equal` tool for testing output, i.e. define an `expected` DataFrame and check it versus `out`.